guides/source/active_record_migrations.md

```diff
@@ -1307,3 +1623,118 @@ special comment like this:
+centralized authority for generating IDs.
+
+#### Enabling UUIDs in Rails
+
+Before using UUIDs in your Rails application, you'll need to ensure that your
+database supports storing them. Additionally, you may need to configure your
+database adapter to work with UUIDs.
+
+1. Database Configuration
+
+    Ensure that your database supports storing UUIDs. PostgreSQL natively
+    supports UUIDs with its `uuid` data type, while other databases might
+    require additional configuration.
+
+    Enabling `uuid-ossp` Extension: If you're using PostgreSQL, enable the
+    `uuid-ossp` extension to make UUID generation functions available. You can
+    do this by adding a migration to enable the extension:
+
+    ```ruby
+    class EnableUuidOsspExtension < ActiveRecord::Migration[8.0]
+      def change
+        enable_extension "uuid-ossp"
+      end
+    end
+    ```
+
+2. Rails Configuration
+
+    In your Rails application configuration file (`config/application.rb`), add
+    the following line to configure Rails to generate UUIDs as primary keys by
+    default:
+
+    ```ruby
+    config.generators do |g|
+      g.orm :active_record, primary_key_type: :uuid
+    end
+    ```
+
+    This setting instructs Rails to use UUIDs as the default primary key type
+    for ActiveRecord models.
+
+3. Adding References with UUIDs:
+
+    When creating associations between models using references, ensure that you
+    specify the data type as :uuid to maintain consistency with the primary key
+    type. For example:
+
+    ``` ruby
+    create_table :posts, id: :uuid do |t|
+      t.references :author, type: :uuid, foreign_key: true
+      # Other columns...
+      t.timestamps
+    end
+    ```
+
+    In this example, the `author_id` column in the posts table references the
+    `id` column of the authors table. By explicitly setting the type to `:uuid`,
+    you ensure that the foreign key column matches the data type of the primary
+    key it references. Adjust the syntax accordingly for other associations and
+    databases.
+
+4. Migration Changes
+
+    When generating migrations for your models, you'll notice that it secifies
+    the id to be of type uuid:
+
+    ```bash
+      $ bin/rails g migration CreateAuthors
+    ```
+
+    ```ruby
+    class CreateAuthors < ActiveRecord::Migration[8.0]
+      def change
+        create_table :authors, id: :uuid do |t|
+          t.timestamps
+        end
+      end
+    end
+    ```
+
+    which results in the following schema:
+
+    ```ruby
+      create_table "authors", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
+      t.datetime "created_at", precision: 6, null: false
+      t.datetime "updated_at", precision: 6, null: false
+    end
+    ```
+
+    In this migration, the `id` column is defined as a UUID primary key with a
+    default value generated by the `gen_random_uuid()` function provided by the
+    `uuid-ossp` extension.
```

Comment:
Slight errata, but the circumstances are confusing and it's easy to understand how you arrived at it. 😄

The `uuid_generate_v4()` function is provided by [`uuid-ossp`](https://www.postgresql.org/docs/current/uuid-ossp.html), but `gen_random_uuid()` was added as a baseline function in [Postgres v13](https://www.postgresql.org/docs/13/functions-uuid.html) specifically to avoid having to use an extension to generate a random one since the `UUID` data type is baseline and often used for its property of randomness.

Reply:
Thanks for the clarification!

I've amended it as follows since the UUID generation functions are no longer needed through the external extensions since Postgres version 13.

```
 In this migration, the `id` column is defined as a UUID primary key with a default value generated by the `gen_random_uuid()` function.
```